frontends/api/src/test-utils/factories/learningResources.ts

```diff
@@ -127,56 +143,91 @@ const learningResource: Factory<LearningResource> = (
     topics: repeat(learningResourceTopic),
     learning_path_parents: [],
     user_list_parents: [],
-    ...typeSpecificOverrides(resourceType),
-    ...overrides,
   }
+}
 
-  function typeSpecificOverrides(type: string): Partial<LearningResource> {
-    if (type === ResourceTypeEnum.Course) {
-      return {
-        platform: faker.lorem.word(),
-        runs: repeat(learningResourceRun, { min: 1, max: 5 }),
-        certification: faker.lorem.word(),
-        offered_by: faker.lorem.word(),
-        course: {
-          extra_course_numbers: maybe(() => repeat(faker.lorem.word)) ?? null,
-        },
-      }
-    } else if (type === ResourceTypeEnum.Program) {
-      return {
-        platform: faker.lorem.word(),
-        certification: faker.lorem.word(),
-        offered_by: faker.lorem.word(),
-      }
-    } else if (type === ResourceTypeEnum.LearningPath) {
-      return {
-        learning_path: {
-          id: faker.unique(faker.datatype.number),
-          item_count: faker.datatype.number({ min: 1, max: 30 }),
-          author: faker.datatype.number(),
-        },
-        learning_path_parents: [],
-      }
-    }
-    return {}
+const learningResource: PartialFactory<LearningResource> = (
+  overrides = {},
+): LearningResource => {
+  overrides = mergeOverrides(
+    {
+      resource_type: learningResourceType(),
+    },
+    overrides,
+  )
+  switch (overrides.resource_type) {
+    case ResourceTypeEnum.Program:
+      // @ts-expect-error TODO
+      return program(overrides)
```

Comment:
Could you add a better note here? Something about the individual resource types (ProgramResource, CourseResource...) having a `resource_type` wider than it should be.